main/src/cgeo/geocaching/utils/MapMarkerUtils.java

```diff
@@ -219,88 +347,76 @@ private static void addListMarkers(final Resources res, final InsetsBuilder inse
+        if (StringUtils.isNotBlank(geocode)) {
+            final Geocache cache = DataStore.loadCache(geocode, LoadFlags.LOAD_CACHE_OR_DB);
+            if (null != cache) {
+                if (cache.isDisabled() || cache.isArchived()) {
+                    DrawableCompat.setTint(dotIcon, ResourcesCompat.getColor(res, R.color.cacheType_disabled, null));
+                }
+            }
+        }
+
+        final InsetsBuilder insetsBuilder = new InsetsBuilder(res, dotMarker.getIntrinsicWidth(), dotMarker.getIntrinsicHeight());
+        insetsBuilder.withInset(new InsetBuilder(dotMarker));
+        insetsBuilder.withInset(new InsetBuilder(dotBackground, Gravity.CENTER));
+        insetsBuilder.withInset(new InsetBuilder(dotIcon, Gravity.CENTER));
+        return buildLayerDrawable(insetsBuilder, 2, 2);
     }
 
     /**
-     * Clear the cache of drawable items.
+     * Provide the LayerDrawable representing the cache type icon
+     *
+     * @param res   Android Resources
+     * @param type  CacheType to get the icon for
+     * @return  Layered Drawable
      */
-    public static void clearCachedItems() {
+    public static Drawable getCacheTypeMarker(final Resources res, final CacheType type) {
+        final int hashcode = new HashCodeBuilder().append(type).toHashCode();
+
         synchronized (overlaysCache) {
-            overlaysCache.clear();
+            CacheMarker marker = overlaysCache.get(hashcode);
+            if (marker == null) {
+                marker = new CacheMarker(hashcode, createCacheTypeMarker(res, type));
+                overlaysCache.put(hashcode, marker);
+            }
+            return marker.getDrawable();
         }
     }
 
     /**
-     * Build the drawable for a given cache.
+     * Build the layered drawable for a cache type icon using a background color + foreground icon
      *
-     * @param res           the resources to use
-     * @param cache         the cache to build the drawable for
-     * @param cacheListType the current CacheListType or Null
-     * @return a drawable representing the current cache status
+     * @param res   Android Resources
+     * @param type  CacheType to get the icon for
+     * @return  Layered Drawable
      */
-    @NonNull
-    @SuppressWarnings("PMD.NPathComplexity") // method readability will not improve by splitting it up
-    private static LayerDrawable createCacheMarker(final Resources res, final Geocache cache, @Nullable final CacheListType cacheListType, final ArrayList<Integer> assignedMarkers) {
-        final int useEmoji = cache.getAssignedEmoji();
-
-        // background: disabled or not
-        final Drawable marker = ResourcesCompat.getDrawable(res, cache.getMapMarkerId(), null);
-        final InsetsBuilder insetsBuilder = new InsetsBuilder(res, marker.getIntrinsicWidth(), marker.getIntrinsicHeight());
+    public static LayerDrawable createCacheTypeMarker(final Resources res, final CacheType type) {
```

Comment:
Probably, this method should be made private as it is designed for internal usage only while the other method directly above (with caching included) is designed for usage in cgeo.

Reply:
> Probably, this method should be made private as it is designed for internal usage only while the other method directly above (with caching included) is designed for usage in cgeo.

done. I also added caching for dotMarkers while at it